define/types.go

```diff
@@ -100,6 +100,8 @@ type Secret struct {
 
 // BuildOutputOptions contains the the outcome of parsing the value of a build --output flag
 type BuildOutputOption struct {
+	Type     string
+	Attrs    map[string]string
 	Path     string // Only valid if !IsStdout
 	IsDir    bool
```

Comment:
Given the addition of `Type`, this should either be removed or at least deprecated. (I’m not sure which one — is this supposed to be a stable external API?)

Reply:
I hadn't thought about this before, primarily because although `Type` as an attribute wasn't present, it was still being used as a string in most places. It'd be premature for me to comment without exploring its usage.

Reply:
I think this must be deprecated and not removed since this is a public API.

Reply:
@flouthoc sorry but could you share how could one go about deprecating this? thanks

Reply:
@danishprakash I think adding comment above the field is enough. This is not the offical doc but I guess you will find similar text in official doc is enough https://zchee.github.io/golang-wiki/Deprecated/